kani-compiler/src/kani_middle/attributes.rs

```diff
@@ -128,6 +147,55 @@ pub fn extract_harness_attributes(tcx: TyCtxt, def_id: DefId) -> Option<HarnessA
     }
 }
 
+/// Check that any unstable API has been enabled. Otherwise, emit an error.
```

Comment:
This function is doing many things: It's checking that all reachable unstable features have been enabled, emits an error if that's not the case, and emits an error also if it wasn't able to parse `#[kani::unstable]` annotations. Please ensure that the description includes this. It can also be more readable if we refactor error handling for the `Ok(unstable_attr)` case.

Reply:
That's fair... let me cleanup this a bit.

Reply:
TBH, the main problem with this code is that we are not really validating the format of the attributes until the codegen stage. In this case, we will detect issues that may be from a dependency. I think the long term solution is to move the attribute handling to happen before codegen.

There are a few issues with that though. The main one for this case is that we don't even use `kani-compiler` today to build the sysroot. So doing this the "proper" way will actually take some time.